core/services/relay/evm/mercury/queue.go

```diff
@@ -37,9 +42,10 @@ const promInterval = 6500 * time.Millisecond
 type TransmitQueue struct {
 	utils.StartStopOnce
 
-	cond sync.Cond
-	lggr logger.Logger
-	mu   *sync.RWMutex
+	cond         sync.Cond
+	lggr         logger.Logger
+	asyncDeleter asyncDeleter
```

Comment:
Just for neatness I'd suggest making `asyncDeleter` an interface

Reply:
It already is an unexported interface!